File: Prog/Light_more_light/Light.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool* toggle_lights(int n) {
    bool* lights = (bool*)malloc(n * sizeof(bool)); // Initialize all lights to be off

    for (int i = 1; i <= n; ++i) {
        for (int j = i - 1; j < n; j += i) {
            lights[j] = !lights[j]; // Toggle the switch of the jth light
        }
    }

    return lights;
}

int main() {
    int num_bulbs = 10;
    bool* bulb_states = toggle_lights(num_bulbs);
    
    for (int i = 0; i < num_bulbs; ++i) {
        printf("Bulb %d: %s\n", i + 1, (bulb_states[i] ? "On" : "Off"));
    }

    free(bulb_states);

    return 0;
}

File: Prog/Light_more_light/Light.py
def toggle_lights(n):
    lights = [False] * n  # Initialize all lights to be off

    for i in range(1, n + 1):
        for j in range(i - 1, n, i):
            lights[j] = not lights[j]  # Toggle the switch of the jth light

    return lights

# Test the function
num_bulbs = 10
bulb_states = toggle_lights(num_bulbs)
for i, state in enumerate(bulb_states):
    print(f"Bulb {i + 1}: {'On' if state else 'Off'}")
